ui/src/app/views/application/show/workflow/tree/item/application.workflow.item.component.ts

```diff
@@ -319,11 +320,14 @@ export class ApplicationWorkflowItemComponent implements DoCheck {
                 this.oldPipelineStatus = this.workflowItem.pipeline.last_pipeline_build.status;
             }
 
-            if (this.oldPipelineStatus === 'Building' && this.oldPipelineStatus !== this.workflowItem.pipeline.last_pipeline_build.status) {
+            if (this.oldPipelineStatus === 'Building' && this.oldPipelineStatus &&
+                this.oldPipelineStatus !== this.workflowItem.pipeline.last_pipeline_build.status &&
+                    this.oldBranch && this.branch === this.oldBranch) {
```

Comment:
When you have a pipeline building and change the branch filter without this condition it will send a false notification